### PR TITLE
Voting

### PR DESCRIPTION
**Pull request by TijmenC**

### CODE REVIEW

File: Abracadabra/abracadabra_web_app/src/components/SubjectQuestionAnswer.js [hunk 1]
```diff
@@ -1,41 +1,164 @@
...
         console.log(res);
         console.log(res.data);
+        if (res.data.vote == 1|-1){
+        setVote({ ...vote, vote: res.data.vote })
+        setVoted(true)
+        }
       })
-      .catch((error) => {
-        console.log(error.response.data);
+        .catch(() => {
```
Comment (VoidBGS) [inline Abracadabra/abracadabra_web_app/src/components/SubjectQuestionAnswer.js]: It is bad practise not to do anything when you catch an error. You can display a message or show something to inform the user of the error that occurred.

File: Abracadabra/abracadabra_web_app/src/components/SubjectQuestionAnswer.js [hunk 2]
```diff
@@ -1,41 +1,164 @@
...
+  }, [vote.vote]);
+
+  const submitPost =  () => {
+    VotesService.PostVoteAnswer(vote).then((res) => {
+      console.log(res);
+      console.log(res.data);
+      setVoted(true)
+     UpdateVotesAnswers()
+    })
+      .catch(() => {
```
Comment (VoidBGS) [inline Abracadabra/abracadabra_web_app/src/components/SubjectQuestionAnswer.js]: Here also

File: Abracadabra/abracadabra_web_app/src/components/SubjectQuestionAnswer.js [hunk 3]
```diff
@@ -1,41 +1,164 @@
...
+  const handleVoteDelete =  () => {
+    VotesService.DeleteVoteAnswer(answer.id).then((res) => {
+      console.log(res);
+      console.log(res.data);
+     UpdateVotesAnswers()
+     setVoted(false)
+     setRendered(false)
+     setVote({ ...vote, vote: null })
+    })
+      .catch((error) => {
```
Comment (VoidBGS) [inline Abracadabra/abracadabra_web_app/src/components/SubjectQuestionAnswer.js:159]: Here also

File: Abracadabra/abracadabra_web_app/src/components/SubjectQuestionAnswer.js [hunk 4]
```diff
@@ -1,41 +1,164 @@
...
+      });
+  };
+  const handleVotePut =  () => {
+    VotesService.PutVoteAnswer(vote).then((res) => {
+      console.log(res);
+      console.log(res.data);
+    UpdateVotesAnswers()
+    })
+      .catch((error) => {
```
Comment (VoidBGS) [inline Abracadabra/abracadabra_web_app/src/components/SubjectQuestionAnswer.js:170]: And here

**Review by VoidBGS — APPROVED** [review]
Looks good, after the addition of error handling.

**Review by Snechar — APPROVED** [review]

**Review by RobbeBryssinck — APPROVED** [review]

**Review by dennycox — APPROVED** [review]

**Review by LarsvdBrandt — APPROVED** [review]

**Pull request merged**